pkg/credstore/credstore.go

```diff
@@ -50,6 +51,43 @@ func Clear(key string) error {
 	return ring.Remove(key)
 }
 
+func ClearWithProfileName(key string) error {
```

Comment:
this will not work for all root profiles which share the same start URL, I would propose that you do this based on the start  URL rather than any profile name